src/main/java/evaluation/listeners/IGameListener.java

```diff
@@ -0,0 +1,113 @@
+package evaluation.listeners;
+
+import core.Game;
+import core.interfaces.IStatisticLogger;
+import evaluation.metrics.*;
+import utilities.Utils;
+
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public interface IGameListener {
+
+    /**
+     * Manages all events.
+     * @param event  Event has information about its type and data fields for game, state, action and player.
+     *               It's not guaranteed that the data fields are different to null, so a check is necessary.
+     */
+    void onEvent(Event event);
+
+
+    /**
+     * This is called when all processing is finished, for example after running a sequence of games
+     * As such, no state is provided.
+     * <p>
+     * This is useful for Listeners that are just interested in aggregate data across many runs
+     */
+    void allGamesFinished();
+
+
+    void setGame(Game game);
+
+    Game getGame();
+
+    /**
+     * Create listener based on given class, logger and metrics class. TODO: more than 1 metrics class
+     * @param listenerClass - class of listener, full path (e.g. evaluation.metrics.MetricsGameListener)
+     * @param logger - class of logger (e.g. evaluation.loggers.SummaryLogger)
+     * @param metricsClass - class of metrics, full path (e.g. evaluation.metrics.GameMetrics)
+     * @return - GameListener instance to be attached to a game
+     */
+    static IGameListener createListener(String listenerClass, IStatisticLogger logger, String metricsClass) {
```

Comment:
This creates a metrics listener specifically. Should either rename method to be clear, or leave this static on the metrics listener class.

Reply:
Good point. Done.